presto-parquet/src/main/java/com/facebook/presto/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -352,4 +472,316 @@ private static void failWithCorruptionException(boolean failOnCorruptedParquetSt
+        {
+            return getValuesAsSlice(type, columnIndex.getMaxValues().size(), columnIndex.getMaxValues());
+        }
+
+        private List<Long> getValuesAsLong(Type type, String column, int pageCount, List<ByteBuffer> values)
+        {
+            List<Long> ret = new ArrayList<>();
+            if (TINYINT.equals(type) || SMALLINT.equals(type) || INTEGER.equals(type)) {
+                for (int i = 0; i < pageCount; i++) {
+                    ret.add((long) converter.convert(values.get(i), column));
+                }
+            }
+            else if (BIGINT.equals(type)) {
+                for (int i = 0; i < pageCount; i++) {
+                    ret.add((long) converter.convert(values.get(i), column));
+                }
+            }
+            else if (REAL.equals(type)) {
+                for (int i = 0; i < pageCount; i++) {
+                    ret.add((long) floatToRawIntBits(converter.convert(values.get(i), column)));
+                }
+            }
+            return ret;
+        }
+
+        private List<Double> getValuesAsDouble(Type type, String column, int pageCount, List<ByteBuffer> values)
+        {
+            List<Double> ret = new ArrayList<>();
+            if (DOUBLE.equals(type)) {
+                for (int i = 0; i < pageCount; i++) {
+                    ret.add(converter.convert(values.get(i), column));
+                }
+            }
+            return ret;
+        }
+
+        private List<Slice> getValuesAsSlice(Type type, int pageCount, List<ByteBuffer> values)
+        {
+            List<Slice> ret = new ArrayList<>();
+            if (isVarcharType(type)) {
+                for (int i = 0; i < pageCount; i++) {
+                    ret.add(Slices.wrappedBuffer(values.get(i)));
+                }
+            }
+            return ret;
+        }
+
+        private <T> T convert(ByteBuffer buf, String name)
+        {
+            return (T) ciConversions.get(name).apply(buf);
+        }
+
+        private Function<Object, Object> getColumnIndexConversions(PrimitiveType type)
+        {
+            switch (type.getPrimitiveTypeName()) {
+                case BOOLEAN:
+                    return buffer -> ((ByteBuffer) buffer).get(0) != 0;
+                case INT32:
+                    return buffer -> ((ByteBuffer) buffer).order(LITTLE_ENDIAN).getInt(0);
+                case INT64:
+                    return buffer -> ((ByteBuffer) buffer).order(LITTLE_ENDIAN).getLong(0);
+                case FLOAT:
+                    return buffer -> ((ByteBuffer) buffer).order(LITTLE_ENDIAN).getFloat(0);
+                case DOUBLE:
+                    return buffer -> ((ByteBuffer) buffer).order(LITTLE_ENDIAN).getDouble(0);
+                case BINARY:
+                case FIXED_LEN_BYTE_ARRAY:
+                case INT96:
+                    return binary -> ByteBuffer.wrap(((Binary) binary).getBytes());
+                default:
```

Comment:
throw exception in default case

Reply:
fixed

Reply:
oh, fine. how about IllegalArgumentException?

Reply:
Sure